ingress/controllers/citrix-netscaler/netscaler/netscaler_resource.go

```diff
@@ -183,11 +183,11 @@ func listBoundResources(resourceName string, resourceType string, boundResourceT
 
 	client := &http.Client{}
 	resp, err := client.Do(req)
-	defer resp.Body.Close()
 	if err != nil {
 		log.Fatal(err)
 		return []byte{}, err
 	} else {
+		defer resp.Body.Close()
```

Comment:
In case of an error "resp" is NULL and using it in the next line causes the controller to crash. The defer is added if there is no error. Otherwise we return.